### ISSUE TITLE
Allow for icon to come from the runtime

### ISSUE THREAD

**Issue by aleixpol** [flatpak/flatpak#842]
As far as I understand it always looks in /app, which is the right thing to do, but we have several applications that will use the icon from the theme and it would be good to accommodate it.

**Comment by alexlarsson** [comment]
You mean the icon in the desktop file?
We don't currently export any files from runtimes, and doing so wouldn't really work with icons, because they would have to be prefixed by the runtime id, just as app exports have to be prefixed by the app.

However, we could have a flatpak-builder feature that lets you easily copy an icon from the sdk into the app. That would solve the problem I think.

**Comment by aleixpol** [comment]
> However, we could have a flatpak-builder feature that lets you easily copy an icon from the sdk into the app. That would solve the problem I think.

I guess that could work. Something like allowing for a full path to be specified in `rename-icon`?

**Comment by alexlarsson** [comment]
@aleixpol I don't think you want that, just specify the icon name and that you want it copied from the runtime, because there could be a whole lot of files to copy (icons in different sizes, high contrast versions, etc).

**Comment by aleixpol** [comment]
Ugh, yes of course.

**Comment by aleixpol** [comment]
Another thing I've noticed is that [dolphin](https://cgit.kde.org/flatpak-kde-applications.git/tree/org.kde.dolphin.json) [desktop file](https://cgit.kde.org/dolphin.git/tree/src/org.kde.dolphin.desktop#n46) has `Icon=system-file-manager` which isn't installed by the project but it still builds and works. Why is that?

**Comment by alexlarsson** [comment]
I think we change the icon name mismatches from an error to a warning, maybe that is why.
